File: updater/src/main/java/identifiers/SecureRandomCallable.java
package identifiers;

import utility.ClassWrapper;

public class SecureRandomCallable extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.getInterfaces().size() == 1 &&
                classNode.getInterfaces().get(0).equals("java/util/concurrent/Callable") &&
                classNode.getInstanceFields().isEmpty();
    }
}

File: find-names/src/main/java/mapper/identifiers/classes/SecureRandomCallable.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.type
import mapper.wrappers.ClassWrapper
import java.util.concurrent.Callable

class SecureRandomCallable : IdentityMapper.Class() {

    override val predicate = predicateOf<ClassWrapper> { it.interfaces == listOf(Callable::class.type) }
            .and { it.instanceFields.isEmpty() }
}